src/lava/lib/optimization/utils/state_analyzer.py

```diff
@@ -0,0 +1,225 @@
+    ) -> None:
+        if filename is None:
+            ax.get_figure().show()
+        else:
+            ax.get_figure().savefig(filename, bbox_inches="tight", dpi=dpi)
+
+    def plot_cost_timeseries(
+        self, report: SolverReport, filename: str = None, ax: plt.Axes = None
+    ) -> None:
+        """Plot cost through time steps."""
+        cost = self._extract_costs_or_warn(report)
+        if cost is None:
+            return
+        ax = self._line_plot(
+            x=list(range(len(cost))),
+            y=cost,
+            xlabel="Time Step",
+            ylabel="Cost",
+            ax=ax,
+        )
+        self._show_or_save(ax=ax, filename=filename)
+
+    def plot_min_cost_timeseries(
+        self, report: SolverReport, filename: str = None, ax: plt.Axes = None
+    ) -> None:
+        """Plot cumulative min of cost through time steps."""
+        cost = self._extract_costs_or_warn(report)
+        if cost is None:
+            return
+        min_cost = np.minimum.accumulate(cost, axis=0)
+        ax = self._line_plot(
+            x=list(range(len(min_cost))),
+            y=min_cost,
+            xlabel="Time Step",
+            ylabel="Best Cost",
+            ax=ax,
+        )
+        self._show_or_save(ax=ax, filename=filename)
+
+    def plot_cost_distribution(
+        self, report: SolverReport, filename: str = None, ax: plt.Axes = None
+    ) -> None:
+        """Plot distribution of cost."""
+        cost = self._extract_costs_or_warn(report)
+        if cost is None:
+            return
+        ax = self._hist_plot(y=cost, xlabel="Cost", ylim=(-4, 104), ax=ax)
+        self._show_or_save(ax=ax, filename=filename)
+
+    def plot_delta_cost_distribution(
+        self, report: SolverReport, filename: str = None, ax: plt.Axes = None
+    ) -> None:
+        """Plot disribution of cost transitions."""
+        cost = self._extract_costs_or_warn(report)
+        if cost is None:
+            return
+        delta_cost = cost[1:] - cost[:-1]
+        ax = self._hist_plot(
+            y=delta_cost, xlabel="Energy Transition", ylim=(-4, 104), ax=ax
+        )
+        self._show_or_save(ax=ax, filename=filename)
+
+    def plot_num_visited_states(
+        self, report: SolverReport, filename: str = None, ax: plt.Axes = None
+    ) -> None:
+        """Plot the total number of unique states visited through time steps"""
+        states = self._extract_states_or_warn(report)
+        if states is None:
+            return
+        num_unique_states = list(
+            map(
+                lambda i: np.unique(states[:i], axis=0).shape[0],
+                range(len(states)),
+            )
+        )
+        ax = self._line_plot(
+            x=list(range(len(states))),
+            y=num_unique_states,
+            xlabel="Time Step",
+            ylabel="# of Visited States",
+            ax=ax,
+        )
+        self._show_or_save(ax=ax, filename=filename)
+
+    def plot_successive_states_distance(
```

Comment:
name is unclear for outsiders. What about plot_state_transition_distance or even plot_states_hamming_distance

Reply:
Nice suggestion!